netty/src/main/java/io/grpc/netty/ProtocolNegotiators.java

```diff
@@ -133,44 +136,86 @@ private SslHandler sslHandler(ChannelHandlerContext ctx) {
    * be negotiated, the {@code handler} is added and writes to the {@link io.netty.channel.Channel}
    * may happen immediately, even before the TLS Handshake is complete.
    */
-  public static ProtocolNegotiator tls(final SslContext sslContext,
-                                       String authority) {
+  public static ProtocolNegotiator tls(SslContext sslContext, String authority) {
     Preconditions.checkNotNull(sslContext, "sslContext");
-    final URI uri = GrpcUtil.authorityToUri(Preconditions.checkNotNull(authority, "authority"));
+    URI uri = GrpcUtil.authorityToUri(Preconditions.checkNotNull(authority, "authority"));
+    return tlsInternal(sslContext, uri.getHost(), uri.getPort());
+  }
 
-    return new ProtocolNegotiator() {
-      @Override
-      public Handler newHandler(Http2ConnectionHandler handler) {
-        ChannelHandler sslBootstrap = new ChannelHandlerAdapter() {
-          @Override
-          public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
-            SSLEngine sslEngine = sslContext.newEngine(ctx.alloc(), uri.getHost(), uri.getPort());
```

Comment:
I have simplified this a bit.  I would prefer to keep the checking -1 as a sanity check.  It wasn't so much about checking that the inputs to SSL stuff are correct, but more that the outputs of URI calls are correct.

I can remove the check if you would still like me too.